tools/astarte_e2e/lib/astarte_e2e/client.ex

```diff
@@ -207,6 +207,29 @@ defmodule AstarteE2E.Client do
     {:via, Registry, {Registry.AstarteE2E, {:client, realm, device_id}}}
   end
 
+  defp notify_service_down(reason) do
+    case ServiceNotifier.notify_service_down(reason) do
+      :mail_sent ->
+        Logger.info("Service down. The user has been notified.", tag: "mail_sent")
```

Comment:
Given all the notes on `notify_service_down` and `notify_service_up`, this is the only Log message that would be printed. So I would just move this into the Notifier (just after `Mailer.deliver_later`), remove the `notify_service_down` and `notify_service_up` wrappers and just call `ServiceNotifier.notify_service_*` directly in the Client